components/x-content-block/src/Timestamp.jsx

```diff
@@ -0,0 +1,35 @@
+import { h } from '@financial-times/x-engine';
+import styles from './ContentBlock.scss';
+
+// this function extracts time in HH:mm format from publishedTimestamp
+function getTimeString(date) {
+	// get hours, minutes and seconds into an array
+	const parts = date.toLocaleTimeString().split(':');
+	// return 'HH:mm'
+	return `${parts[0]}:${parts[1]}`;
+}
+
+export default ({ publishedTimestamp }) => {
+	const now = new Date();
+	const date = new Date(publishedTimestamp.iso);
+	const oneDay = 24 * 60 * 60 * 1000;
+
+	// don't display time string if the post is older than one day
+	// because it is already included in the formatted timestamp
+	const timeString = (now.getTime() - date.getTime() < oneDay) ?
+		getTimeString(date) :
+		'';
+
+	return (
+		<div className={styles['content-block__timestamp-container']}>
+			<time
+				data-o-component="o-date"
+				className={`o-date ${styles['content-block__timestamp']}`}
+				itemProp="datePublished"
+				data-o-date-format={publishedTimestamp.format}
+				dateTime={publishedTimestamp.iso}>{publishedTimestamp.formatted}
+			</time>
+			<span className={styles['content-block__timestamp-exact-time']}>{timeString}</span>
+		</div>
+	);
+};
```

Comment:
I agree that it makes sense to have the logic in a single place.

Reply:
@simonplend I duplicated the time formatting logic here. However, I can't remove it from next-article yet, because the old version still uses that logic. 

I will add a comment in next-article as a reminder.

Reply:
@tbergmen That sounds good :+1:

I think the code in https://github.com/Financial-Times/next-article/pull/3821 which uses that existing logic can now be changed to something like...

```javascript
const publishedTimestamp = (new Date(Number(post.emb) * 1000)).toISOString();
```

...to provide `x-content-block` with an ISO formatted date string.

Reply:
👍 good catch.

I thought that was used by the old version as well but I realised it is specific to x-content-block.

Reply:
It is [used by the old version](https://github.com/Financial-Times/next-article/blob/a9d50f1172ae6cba9e4b567183dc4a94276a2e90/server/model/live-blog/get-posts.js#L2), but the [new version](https://github.com/Financial-Times/next-article/pull/3821) doesn't need to use it any more now that you've moved that logic in to this component.

Reply:
Actually, the new version is also using it when rendering the the posts for the first time at the server side. I prefer to leave it like this until we remove old logic completely.

Reply:
The new version is using it on the server side and the client side, which seems unnecessary now you've added the logic in this component, however you're right that these changes can happen later if that's your preference. I think you'd need to change the timestamp formatting logic in this component back to how it was.

Reply:
We don't need to revert the rendering logic in here, because it is still valid and it is good to have it in one place only.

The problem with changing the structure of `publishedTimestamp` from object to string is that it is used in handlebars to pass the data to the templates. Old and new version are both using handlebars to render blog posts. 

We can still change it by adding another logic like `if (useXContentBlockFlag == true) { use string } else { use object }` when preparing the variables for handlebars. I think it will add an unnecessary complexity to the code that will be removed soon.

Reply:
Oh! The existing Handlebars templates! That had completely slipped my optimistic mind. This all makes sense now - thanks for explaining :+1:

Reply:
As I look deeper into this, I see that the separation of logic for rendering old version and new version via handlebars is already done for other reasons. 

So I decided to go ahead with changing of publishedTimestamp from object to string. A commit will come soon.